Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -1,34 +1,223 @@
-/// Pose2 class is the new Swift type for the SE(2) manifold of 2D Euclidean
-/// Poses
-public struct Pose2: Equatable, Differentiable, KeyPathIterable, TangentStandardBasis {
-  public var t_: Point2
-  public var rot_: Rot2
+import TensorFlow
 
+/// SE(2) Lie group of 2D Euclidean Poses.
+///
+/// We adopt the conventions in [1] and [2] for the tangent space of this manifold and for
+/// derivatives of functions to/from this manifold. In particular:
+///
+///   1. The tangent space at every point is the Lie algebra so(2).
+///   2. For Lie groups `G`, `H`, with algebras `g`, `h`, the differential of any differentiable
+///      function `f : G -> H` at `a \in G` is the linear map `df_a : g -> h` such that
+///        f(a * exp(eps)) ~= f(a) * exp(df_a(eps))
+///      where `eps` is a "small" element of `g`. (And the pullback is the dual map of `df_a`).
+///
+/// (1) is accomplished simply by defining the `TangentVector` type of `Pose2` to be so(2).
+///
+/// (2) is accomplished for `Pose2` by defining differentials[3] satisfying (2) for three
+//  functions:
+///   - .t: SE(2) -> R^2
+///   - .rot: SE(2) -> SO(2)
+///   - init: R^2 x SO(2) -> SE(2)
+/// Since these are the only functions exposed by the original `Pose2` declaration, all other
+/// functions involving `Pose2` are forced to be compositions of these. Therefore, the derivatives
+/// synthesized by Swift's automatic differentiation transform satisfy (2).
+///
+/// [1]: https://github.com/borglab/gtsam/blob/develop/doc/LieGroups.pdf
+/// [2]: https://github.com/borglab/gtsam/blob/develop/doc/math.pdf
+/// [3]: Actually, we define the pullbacks because Swift doesn't support differentials very well
+///      yet.
+public struct Pose2: Differentiable, TangentStandardBasis, KeyPathIterable {
+  /// The pose's translation.
+  ///
+  /// This is the projection SE(2) -> R^2. (Note: not a group homomorphism!)
+  @differentiable public var t: Point2 { tStorage }
+
+  /// The pose's rotation.
+  ///
+  /// This is the projection SE(2) -> SO(2).
+  @differentiable public var rot: Rot2 { rotStorage }
+
+  /// Creates a `Pose2` with rotation `r` and translation `t`.
+  ///
+  /// This is the bijection SO(2) x R^2 -> SE(2), where "x" means direct product of groups. (Note:
+  /// not a group homomorphism!)
   @differentiable
-  public init(_ x: Double, _ y: Double, _ theta: Double) {
-    t_ = Point2(x, y)
-    rot_ = Rot2(theta)
+  public init(_ t: Point2, _ r: Rot2) {
```

Comment:
Let's do (2). I like the idea of making the order in Pose2 and Pose3 consistent with each other. I guess this will make all our Pose2 vectors, jacobians, etc, inconsistent with the GTSAM ones, which is unfortunate.

I'll update this PR to do (2).